sycl/source/detail/event_impl.cpp

```diff
@@ -250,16 +250,25 @@ void event_impl::cleanupCommand(
     detail::Scheduler::getInstance().cleanupFinishedCommands(std::move(Self));
 }
 
+void event_impl::checkProfilingPreconditions() const {
+  if (MQueue.lock() != nullptr) {
```

Comment:
With this profiling operations are always allowed if queue has been deleted. An alternative solution would be to check during construction and storing whether or not profiling is enabled. To reduce property look-up even further you could check once in `queue_impl` and make `event_impl` copy the result of that one check.

Reply:
I got your point. I'll implement it.